src/pages/nft/claim/index.vue

```diff
@@ -762,107 +990,153 @@ export default {
+        'ClaimViewContentDirect',
         this.primaryKey,
         1
       );
     },
-    handleClickView() {
-      logTrackerEvent(this, 'NFT', 'nft_claim_view_button_clicked', '', 1);
-      if (this.collectionId) {
-        if (this.claimingAddress) {
-          this.$router.push(
-            this.localeLocation({
-              name: 'id',
-              params: { id: this.claimingAddress },
-            })
-          );
-        } else {
-          this.$router.push(
-            this.localeLocation({
-              name: 'nft-collection-collectionId',
-              params: {
-                collectionId: this.collectionId,
-              },
-            })
-          );
-        }
-      } else {
-        this.$router.push(
-          this.localeLocation({
-            name: this.nftId ? 'nft-class-classId-nftId' : 'nft-class-classId',
-            params: {
-              classId: this.classId,
-              nftId: this.nftId,
-            },
-          })
-        );
-      }
-    },
-    handleClickRetry() {
-      logTrackerEvent(this, 'NFT', 'nft_claim_retry_button_clicked', '', 1);
-      this.claim();
-    },
-    handleClickViewContentDirectly(e, contentUrl, type) {
+    handleClickNext() {
       logTrackerEvent(
         this,
         'NFT',
-        'ClaimViewContentDirect',
+        `nft_claim_click_next_from_${this.state}`,
         this.primaryKey,
         1
       );
+      switch (this.state) {
+        case NFT_CLAIM_STATE.WELCOME:
+          if (this.claimingAddress) {
+            this.state = NFT_CLAIM_STATE.ID_CONFIRMATION;
+            break;
+          }
+          this.state = NFT_CLAIM_STATE.LOGIN;
+          break;
+
+        case NFT_CLAIM_STATE.LOGIN:
+          this.state = NFT_CLAIM_STATE.ID_CONFIRMATION;
+          break;
+
+        case NFT_CLAIM_STATE.ID_CONFIRMATION:
+          this.state = NFT_CLAIM_STATE.MESSAGE;
+          break;
+
+        case NFT_CLAIM_STATE.MESSAGE:
+          this.state = NFT_CLAIM_STATE.CLAIMING;
+          break;
+
+        case NFT_CLAIM_STATE.CLAIMING:
+          this.state = NFT_CLAIM_STATE.CLAIMED;
+          break;
+
+        default:
+          break;
+      }
     },
-    handleClickNext() {
-      this.state = NFT_CLAIM_STATE.INITIAL;
-    },
-    handleRecommendedItemClick(classId) {
+    async handleClickSignUp() {
       logTrackerEvent(
         this,
         'NFT',
-        'nft_claim_recommend_item_click',
-        classId,
+        `nft_claim_click_sign_up`,
+        this.primaryKey,
         1
       );
+      this.isLoginLoading = true;
+      if (!this.getAddress) {
+        const isConnected = await this.connectWallet({
+          isOpenAuthcore: true,
+        });
+        if (isConnected || this.loginAddress) {
+          this.state = NFT_CLAIM_STATE.ID_CONFIRMATION;
+        }
+        this.isLoginLoading = false;
+      } else {
+        await this.initIfNecessary();
+      }
+      this.isLoginLoading = false;
```

Comment:
Just use 1 `this.isLoginLoading = false;` will do?

Reply:
[🔥 Remove used code](https://github.com/likecoin/liker-land/pull/1588/commits/857fc127de57cb34b7165f4c60c42517546f885f)